src/mesh/unstructured/unstructuredMesh.cpp

```diff
@@ -105,4 +105,105 @@ UnstructuredMesh createSingleCellMesh(const Executor exec)
         boundaryMesh
     );
 }
+
+/** @brief A factory function for a 1D mesh
+ *
+ * A 1D mesh in 3D space in which each cell has a left and a right face.
+ * The 1D mesh is aligned with the x coordinate of Cartesian coordinate system.
+ */
+UnstructuredMesh create1DUniformMesh(const Executor exec, const size_t nCells)
+{
+    const Vector leftBoundary = {0.0, 0.0, 0.0};
+    const Vector rightBoundary = {1.0, 0.0, 0.0};
+    scalar meshSpacing = (rightBoundary[0] - leftBoundary[0]) / nCells;
+    vectorField meshPoints(exec, nCells + 1, {0.0, 0.0, 0.0});
+    meshPoints[0] = leftBoundary;
+    meshPoints[nCells] = leftBoundary;
```

Comment:
i think this is a mistake
```suggestion
    meshPoints[nCells] = rightBoundary;
```